src/main/java/de/tum/in/www1/artemis/service/FileService.java

```diff
@@ -912,32 +910,47 @@ public void replaceVariablesInFileRecursive(String startPath, Map<String, String
      *
      * @param filePath     the path where the file is located
      * @param replacements the replacements that should be applied
-     * @throws IOException if an issue occurs on file access for the replacement of the variables.
      */
-    public void replaceVariablesInFile(String filePath, Map<String, String> replacements) throws IOException {
+    public void replaceVariablesInFile(Path filePath, Map<String, String> replacements) {
         log.debug("Replacing {} in file {}", replacements, filePath);
         // https://stackoverflow.com/questions/3935791/find-and-replace-words-lines-in-a-file
-        Path replaceFilePath = Path.of(filePath);
-        Charset charset = StandardCharsets.UTF_8;
 
-        String fileContent = Files.readString(replaceFilePath, charset);
-        for (Map.Entry<String, String> replacement : replacements.entrySet()) {
-            fileContent = fileContent.replace(replacement.getKey(), replacement.getValue());
+        try {
+            if (isBinaryFile(filePath)) {
+                // do not try to read binary files with 'readString'
+                return;
+            }
+            // Note: we cannot check if this is a binary file or not. We try to read it, in case it fails, we only log this below, but continue
+            String fileContent = Files.readString(filePath, UTF_8);
+            for (Map.Entry<String, String> replacement : replacements.entrySet()) {
+                fileContent = fileContent.replace(replacement.getKey(), replacement.getValue());
+            }
+            Files.writeString(filePath, fileContent, UTF_8);
+        }
+        catch (IOException ex) {
+            log.warn("Exception {} occurred when trying to replace {} in (binary) file {}", ex.getMessage(), replacements, filePath);
+            // continue
         }
-        Files.writeString(replaceFilePath, fileContent, charset);
+    }
+
+    private static boolean isBinaryFile(Path filePath) {
+        // TODO: extend the list of potential binary files
+        var fileName = filePath.toString().toLowerCase();
+
+        return fileName.endsWith(".png") || fileName.endsWith(".jpg") || fileName.endsWith(".heic");
```

Comment:
I'd suggest retrieving the filename from the path and then the extension using FileUtils.
Then, we can easily compare it to a static List that we can define above.

Reply:
this was just a quick implementation, do you have a concrete suggestion how to implement this? if yes, feel free to provide it, otherwise I would move this to a follow-up to deploy the bugfix asp

Reply:
@julian-christl I added your suggestion in https://github.com/ls1intum/Artemis/pull/7178/commits/26b55d71482cbf3f2cff4f2c354427bd8d30ad33 
Is that what you envisioned?

Reply:
Exactly what I envisioned, thank you!